build-tools/cmake/xa_macros.cmake

```diff
@@ -46,13 +46,14 @@ macro(xa_common_prepare)
     Wall
     )
 
-  if(NOT MINGW AND NOT WIN32)
-    # -flto leaves a lot of temporary files with mingw builds, turn the optimization off as we don't really need it there
-    set(XA_COMPILER_FLAGS ${XA_COMPILER_FLAGS} flto)
-  endif()
+# Breaks LLDB debugging
```

Comment:
Do you know **why** it breaks debugging? Is the only way to make it work to turn LTO off? Can you elaborate a bit more on this?

Reply:
No idea why, lto changes the format of the intermediary objects and that confuses the hell out of `dsymutil` who cannot identify debug symbols anymore. I didn't investigate further.

Reply:
Can you make it conditional on `APPLE` then, please?